CONTRIBUTING.md

```diff
@@ -9,23 +9,32 @@ There are 4 main ways of contributing to the PyMC project (in descending order o
 * Contributing or improving the documentation (`docs`) or examples (`pymc/examples`)
 * Submitting issues related to bugs or desired enhancements
 
-# Opening issues
+## Opening issues
 
 We appreciate being notified of problems with the existing PyMC code. We prefer that issues be filed the on [Github Issue Tracker](https://github.com/pymc-devs/pymc/issues), rather than on social media or by direct email to the developers.
 
 Please verify that your issue is not being currently addressed by other issues or pull requests by using the GitHub search tool to look for key words in the project issue tracker.
 
 Filter on the ["beginner friendly"](https://github.com/pymc-devs/pymc/issues?q=is%3Aopen+is%3Aissue+label%3A%22beginner+friendly%22) label for issues which are good for new contributors.
 
-# Contributing code via pull requests
+## Etiquiette for code contributions
+When you start working working on an issue, open a `Draft` pull request as soon as you're doing the first steps.
+Make sure to mention the issue number in the PR description or commit message.
+This way others can see that the issue is being worked on.
+
+Comments like "I want to work on this issue" will be ignored.
+This is because we value getting things done over pledges and found that waiting weeks for follow-ups to such pledges slows down the progress.
+Similarly, we may decide to close or take over pull requests where the author is not responsive.
```

Comment:
This paragraph feels a bit harsh and I think is prone to be misunderstood. Some suggestions:

Issues are not assigned nor reserved via issue comments. Our way of avoiding duplicated work is with draft PRs and
by not opening PRs that overlap with other _active PRs_. We will still track and be active answering any questions posted
on issues to help you open a PR for it.

_Active PRs_ means currently open PRs whose author has commented on or commited in the last week. Once a PR becomes inactive, core contributors can choose to take over or close the PR. The decision will depend on each case and the availability of core contributors and will not necessarly happen exactly on the 7th day of inactivity.

Reply:
see https://github.com/pymc-devs/pymc/pull/5611#discussion_r829891767 first

Reply:
> At the same time, I am disillusioned about lengthy rules and definitions.

They should not be excessively long, but they should be explicit and clear. Now that I have some experience in open source I often skip them (if I am fixing a typo in their docs for example) but bad contributing guides have driven me away from more fleshy PRs in some libraries.

> The most important thing is that we agree on the procedure.

200% agreed on this, if we don't agree on this we'll end up giving incoherent advise which can be even worse than having no docs